<launch>
    <!-- Parameter -->
    <arg name="update_frequency" default="20.0" />
    <arg name="update_params" default="false" />
    <arg name="default_active" default="true"/>
    <arg name="use_dynamic_reconfigure" default="false" />

    <!-- 0 -> nav_msgs::Odometry -->
    <!-- 1 -> geometry_msgs::PoseWithCovarianceStamped -->
    <!-- Remember to change the topic name. -->
    <arg name="odom_type" default="0" />

    <!-- Total Timeout = min_timeout < [(Distance to Goal) * (timeout_a) + (timeout_b)] < max_timeout -->
    <!-- (timeout_a) should be [1 / (robot velocity)] -->
    <arg name="timeout_a" default="100.0" />
    <arg name="timeout_b" default="5.0" />
    <arg name="min_timeout" default="10.0" />
    <arg name="max_timeout" default="100.0" />

    <!-- Resend Goal when the goal is blocked and it isn't timeout yet. -->
    <!-- (1 / resend_goal_frequency) * resend_goal_time = total waiting time for resending goal.  -->
    <arg name="resend_goal_frequency" default="1.0" />
    <arg name="resend_goal_time" default="10" />

    <group ns="navigaton_main">
        <!--  Node  -->
        <node pkg="navigation_main" name="navigation_main_1" type="navigation_main_node" args="navigation_main_1" output="screen">

            <param name="active" type="boolean" value="$(arg default_active)" />
            <param name="timeout_a" type="double" value="$(arg timeout_a)" />
            <param name="timeout_b" type="double" value="$(arg timeout_b)" />
            <param name="min_timeout" type="double" value="$(arg min_timeout)" />
            <param name="max_timeout" type="double" value="$(arg max_timeout)" />
            <param name="update_frequency" type="double" value="$(arg update_frequency)" />
            <param name="use_dynamic_reconfigure" value="$(arg use_dynamic_reconfigure)"/>
            <param name="update_params" value="$(arg update_params)" />
            <param name="odom_type" type="int" value="$(arg odom_type)" />
            <param name="resend_goal_frequency" type="double" value="$(arg resend_goal_frequency)" />
            <param name="resend_goal_time" type="int" value="$(arg resend_goal_time)" />

            <param name="robot_odom_topic" type="string" value="/robot1/odom" />
            <param name="robot_mission_state_topic" type="string" value="/robot1/finishornot" />
            <param name="robot_path_tracker_goal_topic" type="string" value="/robot1/path_tracker_goal" />
            <param name="robot_dock_tracker_goal_topic" type="string" value="/robot1/dock_tracker_goal" />
            <param name="robot_cmd_vel_topic" type="string" value="/robot1/cmd_vel" />
            <param name="robot_path_tracker_cmd_vel_topic" type="string" value="/robot1/path_tracker_cmd_vel" />
            <param name="robot_dock_tracker_cmd_vel_topic" type="string" value="/robot1/dock_tracker_cmd_vel" />
            <param name="main_mission_topic" type="string" value="/robot1/mission" />
            <param name="main_mission_state_topic" type="string" value="/robot1/is_finish" />

        </node>

        <node pkg="navigation_main" name="navigation_main_2" type="navigation_main_node" args="navigation_main_2" output="screen">

            <param name="active" type="boolean" value="$(arg default_active)" />
            <param name="timeout_a" type="double" value="$(arg timeout_a)" />
            <param name="timeout_b" type="double" value="$(arg timeout_b)" />
            <param name="min_timeout" type="double" value="$(arg min_timeout)" />
            <param name="max_timeout" type="double" value="$(arg max_timeout)" />
            <param name="update_frequency" type="double" value="$(arg update_frequency)" />
            <param name="use_dynamic_reconfigure" value="$(arg use_dynamic_reconfigure)"/>
            <param name="update_params" value="$(arg update_params)" />
            <param name="odom_type" type="int" value="$(arg odom_type)" />
            <param name="resend_goal_frequency" type="double" value="$(arg resend_goal_frequency)" />
            <param name="resend_goal_time" type="int" value="$(arg resend_goal_time)" />

            <param name="robot_odom_topic" type="string" value="/robot2/odom" />
            <param name="robot_mission_state_topic" type="string" value="/robot2/finishornot" />
            <param name="robot_path_tracker_goal_topic" type="string" value="/robot2/path_tracker_goal" />
            <param name="robot_dock_tracker_goal_topic" type="string" value="/robot2/dock_tracker_goal" />
            <param name="robot_cmd_vel_topic" type="string" value="/robot2/cmd_vel" />
            <param name="robot_path_tracker_cmd_vel_topic" type="string" value="/robot2/path_tracker_cmd_vel" />
            <param name="robot_dock_tracker_cmd_vel_topic" type="string" value="/robot2/dock_tracker_cmd_vel" />
            <param name="main_mission_topic" type="string" value="/robot2/mission" />
            <param name="main_mission_state_topic" type="string" value="/robot2/is_finish" />

        </node>

    </group>
</launch>